src/_pytest/compat.py

```diff
@@ -76,18 +77,35 @@ def is_async_function(func: object) -> bool:
     return iscoroutinefunction(func) or inspect.isasyncgenfunction(func)
 
 
-def getlocation(function, curdir: Optional[str] = None) -> str:
+class CodeLocation(NamedTuple):
+    path: Path
+    lineno: int
+
+
+def CodeLocation__str__(self: CodeLocation) -> str:
+    """Python 3.6 hack for NamedTuple __str__"""
+    return f"{self.path}:{self.lineno}"
+
+
+setattr(CodeLocation, "__str__", CodeLocation__str__)
+
+
+def getlocation(function, curdir: Optional[Path]) -> CodeLocation:
     function = get_real_func(function)
     fn = Path(inspect.getfile(function))
     lineno = function.__code__.co_firstlineno
+
+    # TODO: this cycle indicates a larger issue
+    from .pathlib import bestrelpath
```

Comment:
Hmm indeed, perhaps `getlocation` can be moved to `pathlib` then? Makes sense there, is all about obtaining the "path" location to a function.